### PR TITLE
fix(snap): move autostart logic to config hook

### PR DESCRIPTION
**Pull request by tonyespy**
Since snapd doesn't make default snap configuration
from the gadget snap available to the install hook,
this change moves the autostart logic to the config
hook, and also modifies the install hook to always
stop/disable the service.

Signed-off-by: Tony Espy <espy@canonical.com>

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-camera-go/blob/master/.github/CONTRIBUTING.md

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

## New Imports

- [ ] Yes
- [ ] No

### CODE REVIEW

File: hooks/cmd/configure/configure.go [hunk 1]
```diff
@@ -63,4 +64,39 @@ func main() {
...
+	case "true":
+		fallthrough
+	case "yes":
+		err = cli.Start("device-camera", true)
+		if err != nil {
+			hooks.Error(fmt.Sprintf("Can't start service - %v", err))
+			os.Exit(1)
+		}
+	case "false":
+	     // no action necessary
```
Comment (siggiskulason) [inline hooks/cmd/configure/configure.go:95]: Setting autostart=true works fine. One thing we might want to document is the fact that setting that value also starts up the service right then, if it is not running already, so if the user wants to set a number of settings before starting up the service, then setting autostart should be the last setting to configure.

**Review by siggiskulason — CHANGES_REQUESTED** [review]
Great work, looks fine, just a couple of minor issues that need fixing. With them changed, I could set autostart values, as well as the env.service.port value and everything behaves as expected.

File: hooks/cmd/configure/configure.go [hunk 2]
```diff
@@ -63,4 +64,39 @@ func main() {
...
+	autostart = strings.ToLower(autostart)
+
+	hooks.Debug(fmt.Sprintf("edgex-device-camera autostart is %s", autostart))
+
+	// service is stopped/disabled by default in the install hook
+	switch autostart {
+	case "true":
+		fallthrough
+	case "yes":
+		err = cli.Start("device-camera", true)
```
[the remark this answers is not in the document]
Comment (tonyespy) [reply hooks/cmd/configure/configure.go]: fixed & force-pushed

[the remark this answers is not in the document]
Comment (siggiskulason) [reply hooks/cmd/configure/configure.go]: @tonyespy - could you plese also fix the typo lin line 45?
i.e.  `hooks.Init(debug, "egex-device-camera-go");` should be  `hooks.Init(debug, "edgex-device-camera-go");`

[the remark this answers is not in the document]
Comment (tonyespy) [reply hooks/cmd/configure/configure.go]: done & pushed again

**Review by siggiskulason — APPROVED** [review]
LGTM

**Pull request merged**